### PR TITLE
Check gcloud for project, and add unit tests.

### PR DESCRIPTION
**Pull request by G-Harmon**
Adds missing functionality for getting GCP Project from gcloud settings, and adds unit tests for this.

cc @nikhiljindal

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/172)

### CODE REVIEW

File: app/kubemci/cmd/getstatus.go [hunk 1]
```diff
@@ -77,7 +79,12 @@ func validateGetStatusArgs(options *GetStatusOptions, args []string) error {
 	}
 	// Verify that the required params are not missing.
 	if options.GCPProject == "" {
```
Comment (nikhiljindal) [inline app/kubemci/cmd/getstatus.go]: You will also need to update the flag description above in addFlags to say this is optional in flag description.

**Comment by nikhiljindal** [comment]
Looks great, One comment about updating flag description to clarify its optional.
Also xref https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/138 for easy referencing :)

**Comment by G-Harmon** [comment]
This is part of fixing #80

File: app/kubemci/cmd/getstatus.go (hunk 1 above)
Comment (G-Harmon) [reply app/kubemci/cmd/getstatus.go]: fixed.

File: app/kubemci/cmd/getstatus.go [hunk 2]
```diff
@@ -66,7 +68,7 @@ func NewCmdGetStatus(out, err io.Writer) *cobra.Command {
 
 func addGetStatusFlags(cmd *cobra.Command, options *GetStatusOptions) error {
 	// TODO(nikhiljindal): Add a short flag "-p" if it seems useful.
-	cmd.Flags().StringVarP(&options.GCPProject, "gcp-project", "", options.GCPProject, "[required] name of the gcp project")
+	cmd.Flags().StringVarP(&options.GCPProject, "gcp-project", "", options.GCPProject, "[optional] name of the gcp project")
```
Comment (nikhiljindal) [inline app/kubemci/cmd/getstatus.go]: Description as copied from that other PR:
```name of the gcp project. Is fetched using gcloud config get-value project if unset here```
Use the same here?

Comment (G-Harmon) [reply app/kubemci/cmd/getstatus.go]: Done. (seems like it'd read a little better as "It's fetched using...", but might as well keep it consistent.)

**Comment by nikhiljindal** [comment]
/lgtm
thx for the fixes

**Comment by nikhiljindal** [comment]
will merge once tests come back green

**Pull request merged**